backend/application/syson-application-configuration/src/main/java/org/eclipse/syson/application/configuration/SysMLEditingContextProcessor.java

```diff
@@ -63,10 +64,14 @@ public void preProcess(IEditingContext editingContext) {
                 && !this.studioCapableEditingContextPredicate.test(editingContext.getId())) {
             // Do not initialize the editing context as a SysON editing context if it contains a studio.
             siriusWebEditingContext.getDomain().getResourceSet().eAdapters().add(new SysONEContentAdapter());
+            // Register an adapter to keep track of all element in immutable library to be able to access them
```

Comment:
Why did you move the instantiation of _targetResourceSet_? I don't see the advantage of moving it.
Unless I'm missing something please move it at it's original place.

Reply:
Because it might cause some stackoverflow issue.

The resources are interdependant and the refrence are not yet handled that can be problematic in the implicit switch.

We need all reference to be set before sing the implicit switch with this class. Not doing some cause infinite loop while checking for exisiting specialization.

Reply:
Because it was called 3 time in this method. But I can also reuse it la 66 so I will done that.